test/bdd/pkg/waci/waci_issuance_sdk_steps.go

```diff
@@ -0,0 +1,715 @@
+	suite.Step(`^"([^"]*)" accepts the Credential Application and sends a credential to the Holder$`,
+		i.acceptCredentialApplication)
+	suite.Step(`^"([^"]*)" accepts the credential$`, i.acceptCredential)
+	suite.Step(`^Holder checks that the expected credential was received in a Credential `+
+		`Fulfillment attachment$`, i.checkCredential)
+}
+
+func (i *IssuanceSDKSteps) createOOBV2WithStreamlinedVCGoalCode(issuerName string) error {
+	issuerOOBV2Client, err := outofbandv2.New(i.context.AgentCtx[issuerName])
+	if err != nil {
+		return fmt.Errorf("failed to create an OOB V2 client for %s: %w", issuerName, err)
+	}
+
+	issuerDIDDoc, ok := i.context.PublicDIDDocs[issuerName]
+	if !ok {
+		return fmt.Errorf("DID document for %s is unexpectedly missing from the context", issuerName)
+	}
+
+	invitation, err := issuerOOBV2Client.CreateInvitation(
+		outofbandv2.WithGoal("", waciIssuanceGoalCode),
+		outofbandv2.WithAccept("didcomm/v2"),
+		outofbandv2.WithFrom(issuerDIDDoc.ID),
+		outofbandv2.WithLabel(issuerName),
+	)
+	if err != nil {
+		return fmt.Errorf("failed to create oob invitation for %s: %w", issuerName, err)
+	}
+
+	i.oobV2InviteFromIssuerToHolder = invitation
+
+	i.context.OutOfBandV2Clients[issuerName] = issuerOOBV2Client
+
+	return nil
+}
+
+func (i *IssuanceSDKSteps) acceptOOBV2Invitation(issuerName, holderName string) error {
+	holderOOBV2Client, err := outofbandv2.New(i.context.AgentCtx[holderName])
+	if err != nil {
+		return fmt.Errorf("failed to create an OOB V2 client for %s: %w", issuerName, err)
+	}
+
+	_, err = holderOOBV2Client.AcceptInvitation(i.oobV2InviteFromIssuerToHolder)
+	if err != nil {
+		return fmt.Errorf("failed to accept OOBV2 invitation from %s to %s : %w", issuerName, holderName, err)
+	}
+
+	return nil
+}
+
+func (i *IssuanceSDKSteps) sendsProposalV3(holderName, issuerName string) error {
+	err := i.createIssueCredentialClients(holderName, issuerName)
+	if err != nil {
+		return err
+	}
+
+	i.createActions(holderName, issuerName)
+
+	err = i.registerActionsAndEvents(holderName, issuerName)
+	if err != nil {
+		return err
+	}
+
+	i.context.DIDExchangeClients[holderName], err = didexClient.New(i.context.AgentCtx[holderName])
+	if err != nil {
+		return err
+	}
+
+	proposeCredentialMsg := issuecredentialclient.ProposeCredential{
+		Type:         issuecredential.ProposeCredentialMsgTypeV3,
+		ID:           uuid.New().String(),
+		InvitationID: i.oobV2InviteFromIssuerToHolder.ID,
+	}
+
+	connection, err := i.GetConnection(holderName, issuerName)
+	if err != nil {
+		return err
+	}
+
+	piid, err := i.issueCredentialClients[holderName].SendProposal(&proposeCredentialMsg, connection.Record)
+	if err != nil {
+		return fmt.Errorf("failed to send proposal: %w", err)
+	}
+
+	if piid == "" {
+		return errors.New("piid is empty")
+	}
+
+	return nil
+}
+
+func (i *IssuanceSDKSteps) acceptProposalV3(issuerName string) error {
+	piid, _, err := i.getActionID(issuerName, false)
```

Comment:
you are not cross checking invitation ID here to co-relate the proposal with the invitation.

you may have to do the similar correlation for other steps too.

Reply:
Added Invitation ID cross checking.

I believe the other steps don't need it, since I use PIID with the issue credential client to accept and continue through the protocol, but let me know if there's something else I missed.